graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java

```diff
@@ -915,6 +919,12 @@ private QueryBuilder standardAggregationFilters(TimeRange range, String filter)
         for (IndexRange indexRange : indexRanges) {
             // if we aren't in a stream search, we look at all the ranges matching the time range.
             if (indexSet == null && filter == null) {
+                // Exclude Event indices by name
+                // TODO This will fail if the prefix settings are changed. But it's the only info we have in IndexRanges
+                if (indexRange.indexName().startsWith(elasticsearchConfiguration.getDefaultEventsIndexPrefix()) ||
```

Comment:
I think we can check if the index is part of an event index set so we don't have to rely on the default events index prefixes.

Something like this could work (untested :smile:):

```diff
diff --git graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java
index 0091396d1..d39f8962a 100644
--- graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java
+++ graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java
@@ -60,6 +60,7 @@ import org.graylog2.indexer.IndexMapping;
 import org.graylog2.indexer.IndexSet;
 import org.graylog2.indexer.IndexSetRegistry;
 import org.graylog2.indexer.cluster.jest.JestUtils;
+import org.graylog2.indexer.indexset.IndexSetConfig;
 import org.graylog2.indexer.indices.Indices;
 import org.graylog2.indexer.ranges.IndexRange;
 import org.graylog2.indexer.ranges.IndexRangeService;
@@ -912,9 +913,21 @@ public class Searches {
 
         final ImmutableSortedSet.Builder<IndexRange> indices = ImmutableSortedSet.orderedBy(IndexRange.COMPARATOR);
         final SortedSet<IndexRange> indexRanges = indexRangeService.find(range.getFrom(), range.getTo());
+        final Set<String> affectedIndexNames = indexRanges.stream().map(IndexRange::indexName).collect(Collectors.toSet());
+        final Set<IndexSet> nonEventIndexSets = indexSetRegistry.getForIndices(affectedIndexNames).stream()
+                .filter(indexSet1 -> !IndexSetConfig.TemplateType.EVENTS.equals(indexSet1.getConfig().indexTemplateType().orElse(IndexSetConfig.TemplateType.MESSAGES)))
+                .collect(Collectors.toSet());
+
         for (IndexRange indexRange : indexRanges) {
             // if we aren't in a stream search, we look at all the ranges matching the time range.
             if (indexSet == null && filter == null) {
+                // Don't include the index range if it's for an event index set to avoid sorting issues.
+                // See the following issues for details:
+                // - https://github.com/Graylog2/graylog2-server/issues/6384
+                // - https://github.com/Graylog2/graylog2-server/issues/6490
+                if (nonEventIndexSets.stream().noneMatch(set -> set.isManagedIndex(indexRange.indexName()))) {
+                    continue;
+                }
                 indices.add(indexRange);
                 continue;
             }
```

Reply:
Hmm, I don't see how this is any different.
https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java#L106

If the user changes the event index name, we will still fail to exclude the old indices that are created before the renaming took place.

Reply:
Ah, nevermind. We would still have the IndexSets with the old names then. :)

Reply:
The difference is, that the user cannot change the index prefix for existing index sets. The prefix can only be set when creating an index set. The default prefix can be changed any time in the config file. I agree that this probably doesn't make a huge difference in practice at the moment. But once we allow users to create event index sets on their own, the code that checks the default event index set prefixes will break.